# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by scripts/Create-CopyPatch.
# 
# T2 SDE: architecture/powerpc64/package/*/0667-ps3fb-cursor.patch
# Copyright (C) 2019 - 2021 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---

--- linux-5.0/drivers/video/fbdev/ps3fb.c.vanilla	2019-07-14 14:53:01.797322098 +0000
+++ linux-5.0/drivers/video/fbdev/ps3fb.c	2019-07-14 15:15:20.841316556 +0000
@@ -987,6 +973,11 @@

 	ps3fb_sync_image(info->device, 0);
 
+
+	/* test cursor init here */
+	int status = lv1_gpu_context_attribute(ps3fb.context_handle, 0x10b, 0/*head*/, 1, 0, 0); /* init */
+	if (status) dev_err(info->device, "%s: cursor init failed (%d)\n", __func__, status);
+
 	return 0;
 }
 
@@ -1265,6 +1256,19 @@
 		}
 		break;
 
+	case PS3FB_IOCTL_CURSOR_ENABLE:
+		retval = lv1_gpu_context_attribute(ps3fb.context_handle, 0x10c, 0/*head*/, arg ? 0x1 : 0x02, 0x0, 0x0);	/* enable/disable */
+		break;
+
+	case PS3FB_IOCTL_CURSOR_POS:
+ 		retval = lv1_gpu_context_attribute(ps3fb.context_handle, 0x10b, 0/*head*/, 3, arg & 0xffff, arg >> 16); /* x/y pos */
+		break;
+
+	case PS3FB_IOCTL_CURSOR_OFFS:
+		retval = lv1_gpu_context_attribute(ps3fb.context_handle, 0x10b, 0/*head*/, 2, arg, 0); /* offset */
+		//retval = lv1_gpu_context_attribute(context_handle, 0x10b, 0/*head*/, 4, 0, 0); /* ??? */
+		break;
+
 	default:
 		retval = -ENOIOCTLCMD;
 		break;
--- linux-5.0/arch/powerpc/include/uapi/asm/ps3fb.h.vanilla	2019-07-14 14:54:38.629321697 +0000
+++ linux-5.0/arch/powerpc/include/uapi/asm/ps3fb.h	2019-07-14 15:11:45.477317448 +0000
@@ -34,6 +34,10 @@
 #define PS3FB_IOCTL_GPU_INFO      _IOR('r', 8, int)	/* get GPU info */
 #define PS3FB_IOCTL_GPU_ATTR      _IOW('r', 9, int)	/* set attribute */
 
+#define PS3FB_IOCTL_CURSOR_ENABLE _IOW('r', 10, int)	/* cursor enable/disable */
+#define PS3FB_IOCTL_CURSOR_POS    _IOW('r', 11, int)	/* cursor x/y pos*/
+#define PS3FB_IOCTL_CURSOR_OFFS   _IOW('r', 12, int)	/* cursor data offset */
+
 #ifndef FBIO_WAITFORVSYNC
 #define FBIO_WAITFORVSYNC         _IOW('F', 0x20, __u32) /* wait for vsync */
 #endif
